teps/0025-proxy-support.md

```diff
@@ -0,0 +1,133 @@
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+The proposal is to provide the proxy support to tektoncd/pipelines and tektoncd/triggers.
+
+Tekton should provide a way to use network proxy settings defined for the platform so that workloads
+can make use of it during execution.
+
+## Motivation
+
+This will enable users/organisations to enable network proxy settings for CI/CD using tekton if their
+workload is expected to communicate to outside world through proxy.
+
+### Goals
+
+1. Workloads executions like taskrun/pipelinerun should use global proxy settings.
+2. Controllers should use global proxy settings for communicating.
+
+### Non-Goals
+
+1. Managing or maintaining proxy settings.
+
+## Proposal
+
+This proposal is to allow tektoncd/pipeline and tektoncd/triggers to have proxy support
+enabled / configurable to benefit users/organizations to use proxy feature to communicate to
+the outside world of the platform.
+
+It can be configured to use proxy setting defined in the platform.
+
+### User Stories (optional)
+
+User should be able to define global proxy settings on the platform and tektoncd/pipeline
+and tektoncd/triggers act based on the proxy settings defined.
+
+1. User should be able to define proxy config for all workloads
+2. User should be able to provide proxy config for specific taskrun/pipelinerun
+3. Controllers should use proxy setting for communication
+
+### Notes/Constraints/Caveats (optional)
+
+1. If a user configures proxy settings for platform, should we provide a way to run some workloads without proxy settings.
+2. Changes required for tektoncd/triggers (yet to explore)
+
+### Risks and Mitigations
+
+None
+
+### User Experience (optional)
+
+There will be some spec changes for providing the proxy settings to pipelinrun/taskrun/pod level.
+
+### Performance (optional)
+
+None
+
+## Design Details
+
+We can provide a standard configmap to define the global proxy settings. User can create a configmap with respective data
+to provide the proxy settings as configmap
+
+Later, controller based on the data available in configmap, it should add the env variables related to
+the global proxy setting at the pipelinrun/taskrun/pod spec.
+
+We should also provide a way to specify env variable to the pipelinerun/taskrun spec so that
+a specific workload should also be run with proxy settings if a user dont want to run the all workloads of environment.
+
+Issues to be resolved
+
+1. [Proxy support in workloads](https://github.com/tektoncd/pipeline/issues/3090)
+2. [Way to set env for workloads](https://github.com/tektoncd/pipeline/issues/1606)
```

Comment:
This is high level and should be under the `Proposal` header. And we should get into detail in here (if needed), to show an example of a configmap, or the changes to the spec, …